storage/storage-client/src/lib.rs

```diff
@@ -398,6 +423,17 @@ pub trait StorageRead: Send + Sync {
         &self,
         start_epoch: u64,
     ) -> Pin<Box<dyn Future<Output = Result<Vec<LedgerInfoWithSignatures>>> + Send>>;
+
+    /// See [`LibraDB::backup_account_state`].
+    ///
+    /// Due to the streaming nature of this API, only an async version is provided.
+    ///
+    /// [`LibraDB::backup_account_state`]:
+    /// ../libradb/struct.LibraDB.html#method.backup_account_state
+    fn backup_account_state_async(
+        &self,
+        version: u64,
+    ) -> Result<BoxStream<Result<BackupAccountStateResponse, Error>>>;
```

Comment:
Do you still prefer to return a stream with a single error v.s. the current API?

Reply:
Error semantics are weird and hard to get right with futures01 streams.  We don't have much control over it for now, AFAICT.